robot_framework/eflyt.py

```diff
@@ -0,0 +1,338 @@
+from OpenOrchestrator.database.queues import QueueStatus
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.select import Select
+
+from robot_framework import config
+
+
+ALLOWED_CASE_TYPES = (
+    "Logivært",
+    "Boligselskab",
+    "For sent anmeldt"
+)
+
+
+def login(orchestrator_connection: OrchestratorConnection) -> webdriver.Chrome:
+    """Opens a browser and logs in to Eflyt.
+
+    Args:
+        orchestrator_connection: The connection to Orchestrator.
+
+    Returns:
+        A selenium browser object.
+    """
+    eflyt_creds = orchestrator_connection.get_credential(config.EFLYT_CREDS)
+
+    browser = webdriver.Chrome()
+    browser.maximize_window()
+    browser.get("https://notuskommunal.scandihealth.net/")
+
+    user_field = browser.find_element(By.ID, "Login1_UserName")
+    user_field.send_keys(eflyt_creds.username)
+
+    pass_field = browser.find_element(By.ID, "Login1_Password")
+    pass_field.send_keys(eflyt_creds.password)
+
+    browser.find_element(By.ID, "Login1_LoginImageButton").click()
+
+    browser.get("https://notuskommunal.scandihealth.net/web/SearchResulteFlyt.aspx")
+
+    return browser
+
+
+def search_cases(browser: webdriver.Chrome) -> None:
+    """Apply the correct filters in Eflyt and search the case list.
+
+    Args:
+        browser: The webdriver browser object.
+    """
+    sagstilstand_select = Select(browser.find_element(By.ID, "ctl00_ContentPlaceHolder1_SearchControl_ddlTilstand"))
+    sagstilstand_select.select_by_visible_text("Ubehandlet")
+
+    search_date = date.today().strftime("%d%m%Y")
+    date_input = browser.find_element(By.ID, "ctl00_ContentPlaceHolder1_SearchControl_txtFlytteEndDato")
+    date_input.send_keys(search_date)
+
+    search_button = browser.find_element(By.ID, "ctl00_ContentPlaceHolder1_SearchControl_btnSearch")
+    search_button.click()
+
+
+def extract_cases(browser: webdriver.Chrome) -> list[str]:
+    """Extract and filter cases from the case table.
+
+    Args:
+        browser: The webdriver browser object.
+
+    Returns:
+        A list of filtered case objects.
+    """
+    table = browser.find_element(By.ID, "ctl00_ContentPlaceHolder2_GridViewSearchResult")
+    rows = table.find_elements(By.TAG_NAME, "tr")
+
+    # remove header row
+    rows.pop(0)
+
+    cases = []
+    for row in rows:
+        case_number = row.find_element(By.XPATH, "td[4]").text
+        case_types_text = row.find_element(By.XPATH, "td[5]").text
+
+        # If the case types ends with '...' we need to get the title instead
+        if case_types_text.endswith("..."):
+            case_types_text = row.find_element(By.XPATH, "td[5]").get_attribute("Title")
+
+        case_types = case_types_text.split(", ")
+
+        # Check if there are any case types other than the allowed ones
+        for case_type in case_types:
+            if case_type not in ALLOWED_CASE_TYPES:
+                break
```

Comment:
`break`'ing here will skip all allowed case types following a disallowed one in `case_types`. `continue` seems to be the right choice here to make sure that you're collecting all allowed cases.

Reply:
This code snippet reads as: "If the case has any types other than the allowed ones, skip it". I'm only breaking out of the type checking loop for the one case.

Reply:
Ahh, it's a `for/else` construct. I read it as an `if/else`.